scripts/api.py

```
@@ -0,0 +1,334 @@
+                break
+
+        api_logger.info(f"File \"{local_file_path}\" as \"{remote_file_path}\" replaced in dataset {self.dataset_doi}.")
+        return response
+
+    def upload_file(self, local_file_path, description: str):
+        """Upload a file to the dataset, checking for duplicates."""
+        existing_file = self.__check_if_file_exists(local_file_path)
+
+        if existing_file:
+            md5_info = existing_file.get('dataFile', {}).get('md5', '')
+            md5_hash = hashlib.md5(open(local_file_path, 'rb').read()).hexdigest()
```

Ruff (0.12.2)

Use a context manager for opening files

(SIM115)